packages/trader/src/AppV2/Components/TradeParameters/RiskManagement/take-profit-and-stop-loss-input.tsx

```diff
@@ -0,0 +1,325 @@
+                    ? localize('Please enter a take profit amount.')
+                    : localize('Please enter a stop loss amount.')
+            );
+            return;
+        }
+
+        const is_tp_enabled = error_text ? false : is_enabled;
+        onChangeMultiple({
+            ...(is_take_profit_input ? { has_take_profit: is_tp_enabled } : { has_stop_loss: is_tp_enabled }),
+            ...(is_take_profit_input
+                ? {
+                      take_profit: error_text || new_input_value === '0' ? '' : new_input_value,
+                  }
+                : {
+                      stop_loss: error_text || new_input_value === '0' ? '' : new_input_value,
+                  }),
+            ...(is_tp_enabled ? { has_cancellation: false } : {}),
+        });
+        onActionSheetClose();
+    };
+
+    React.useEffect(() => {
+        setErrorText(initial_error_text ?? '');
+        updateParentRef({
+            field_name: is_take_profit_input ? 'tp_error_text' : 'sl_error_text',
+            new_value: initial_error_text ?? '',
+        });
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [initial_error_text]);
+
+    React.useEffect(() => {
+        setInfo(info =>
+            (info.min_value !== min_value && min_value) || (info.max_value !== max_value && max_value)
+                ? { min_value, max_value }
+                : info
+        );
+    }, [min_value, max_value]);
+
+    return (
+        <React.Fragment>
+            <Component className={clsx('take-profit__wrapper', classname)}>
+                <div className='take-profit__content'>
+                    <Text>
+                        {is_take_profit_input ? (
+                            <Localize i18n_default_text='Take profit' />
+                        ) : (
+                            <Localize i18n_default_text='Stop loss' />
+                        )}
+                    </Text>
+                    <ToggleSwitch checked={is_enabled} onChange={onToggleSwitch} />
+                </div>
+                <TextFieldWithSteppers
+                    allowDecimals
+                    customType='commaRemoval'
+                    disabled={!is_enabled}
+                    decimals={decimals}
+                    data-testid={is_take_profit_input ? 'dt_tp_input' : 'dt_sl_input'}
+                    inputMode='decimal'
+                    message={error_text || input_message}
+                    minusDisabled={Number(new_input_value) - 1 <= 0}
+                    name={type}
+                    onChange={onInputChange}
+                    placeholder={localize('Amount')}
+                    ref={input_ref}
+                    regex={/[^0-9.,]/g}
+                    status={error_text ? 'error' : 'neutral'}
+                    textAlignment='center'
+                    unitLeft={currency_display_code}
+                    variant='fill'
+                    value={new_input_value}
```

Comment:
let's also add `noStatusIcon` prop (as true) to `TextFieldWithSteppers` which will prevent red ⚠️ icon from showing up when there is an error in the input. this prop is available in the quill-ui version you have installed :)